deep_quoridor/src/quoridor_env.py

```diff
@@ -132,6 +132,25 @@ def _get_observation(self, agent_id):
             "opponent_walls_remaining": self.walls_remaining[self._opponent(agent_id)],
         }
 
+    def _is_wall_between(self, row_a, col_a, row_b, col_b):
+        """
+        Returns True if there is a wall between pos_a and pos_b
+        NOTE: The min and max tricks there are just a lazy way to avoid overindexing
+        Since the wall grid is one smaller than the board (because boards take two spaces)
+        """
+        if row_a == row_b:  # Horizontal movement - check vertical walls
+            wall_col = min(col_a, col_b)
+            return (
+                self.walls[min(row_a, self.wall_size - 1), wall_col, 0] == 1
+                or self.walls[max(row_a - 1, 0), wall_col, 0] == 1
+            )
+        elif col_a == col_b:  # Vertical movement - check horizontal walls
+            wall_row = min(row_a, row_b)
+            return (
+                self.walls[wall_row, min(col_a, self.wall_size - 1), 1] == 1
+                or self.walls[wall_row, max(col_a - 1, 0), 1] == 1
+            )
+
```

Comment:
I'd raise an exception here, because if by accident (e.g. when checking jumps) we pass both a different row and column, then `None` will be returned and may be evaluated as `False`, and will be hard to debug for our other 999 developers

Reply:
Also, we could rise if the difference between rows or cols is other than 1 (e.g. if we accidentally call it with a jump of 2)